opennms-web-api/src/main/java/org/opennms/web/svclayer/support/DefaultGraphResultsService.java

```diff
@@ -154,6 +165,25 @@ public GraphResults findResults(ResourceId[] resourceIds, String[] reports, Stri
                 }
             }
         }
+        // GraphSelected case where all resources are fetched from generatedId
+        if (!Strings.isNullOrEmpty(generatedId) && m_jsonStore != null) {
+            Optional<String> result = m_jsonStore.get(generatedId, RESOURCE_IDS_CONTEXT);
+            if (result.isPresent()) {
+                try {
+                    String[] resourceArray = m_gson.fromJson(result.get(), String[].class);
+                    for (String resourceId : resourceArray) {
+                        try {
+                            OnmsResource resource = m_resourceDao.getResourceById(ResourceId.fromString(resourceId));
+                            graphResults.addGraphResultSet(createGraphResultSet(null, resource, reports, graphResults));
+                        } catch (IllegalArgumentException e) {
```

Comment:
Is this catch block attempting to cover the case where a resource is no longer valid?

Reply:
Fixed